### PR TITLE
Ignore unusable files in helpers directory

### LINKED ISSUE #118: Warning for irrelevant files
Vision is giving me this warning upon initial server startup:

```
WARNING: vision failed to load helper '/Users/sholladay/Code/sandbox/foo-project/lib/view/helper/README.md': Invalid or unexpected token
```

Here is the config:

```js
server.views({
    engines : {
        html : handlebars
    },
    relativeTo   : path.join(__dirname, 'lib', 'view'),
    path         : '.',
    helpersPath  : 'helper',
    partialsPath : 'partial',
    layoutPath   : 'layout',
    layout       : 'default-layout'
});
```

So, I see that Vision [ignores hidden files](https://github.com/hapijs/vision/blob/10760955c69b54450adcaaa219ac2f945c38ee3d/lib/manager.js#L250), which is great. In my case, I have a `README.md` in the helpers directory to provide some documentation. It would be great if Vision could ignore this too.

I think the correct fix would be something along the lines of:

```js
const isRequirable = Object.keys(require.extensions).includes(path.extname(file));
if (!isRequirable) {
    return;
}
```

### PR DESCRIPTION
**Pull request by sholladay**
This PR aims to ensure that warnings like this ...

```
WARNING: vision failed to load helper '/Users/sholladay/Code/sandbox/foo-project/lib/view/helper/README.md': Invalid or unexpected token
```

... do not appear for files that _obviously_ cannot be loaded as a helper, such as Markdown files. It skips loading files whose extension does not appear in `require.extensions`, which is the source of truth for what `require()` knows how to load.

I also tried to make the tests more realistic, with actual `.js` files used for the invalid helper test cases in addition to `.md` and `.foo` for good measure.

Fixes #118

### CODE REVIEW

**Comment by sholladay** [comment]
@jagoda as the lead maintainer, are you able to offer any insight into when this might be able to get a review and/or merge?

**Comment by sholladay** [comment]
@Marsup you are the most recent committer to this project. Any thoughts or suggestions to move this forward?

File: lib/manager.js [hunk 1]
```diff
@@ -246,21 +246,22 @@ internals.Manager.prototype._loadHelpers = function (engine) {
 
             file = Path.join(path, file);
             const stat = Fs.statSync(file);
-            if (!stat.isDirectory() &&
-                Path.basename(file)[0] !== '.') {
+            if (stat.isDirectory() || Path.basename(file)[0] === '.' ||
+                Object.keys(require.extensions).indexOf(Path.extname(file)) === -1) {
```
Comment (wswoodruff) [inline lib/manager.js]: Can we make this conditional multiline?
```
if (stat.isDirectory() ||
    Path.basename(file)[0] === '.' ||
    Object.keys(require.extensions).indexOf(Path.extname(file)) === -1) {
    ...
}
```

Comment (sholladay) [reply lib/manager.js]: Sure. Unfortunately, there are merge conflicts since this was left for so long.

I'll look into it.

**Comment by wswoodruff** [comment]
Let's ensure the tests are passing before merging

File: test/manager.js [hunk 2]
```diff
@@ -1534,8 +1534,14 @@ describe('Manager', () => {
...
-            expect(output).to.contain('invalid/helpers/bad1.module');
-            expect(output).to.contain('invalid/helpers/bad2.module');
+
+            expect(output).to.contain('invalid/helpers/bad1.js');
+            expect(output).to.contain('invalid/helpers/bad1.json');
+
+            // Ignore non-requirable file extensions
+            expect(output).to.not.contain('invalid/helpers/bad1.foo');
+            expect(output).to.not.contain('invalid/helpers/README.md');
+            done();
```
Comment (wswoodruff) [inline test/manager.js]: Remove this line and we're good to go

**Comment by sholladay** [comment]
Should be good to go now @wswoodruff, PTAL.

File: test/manager.js (hunk 2 above)
Comment (sholladay) [reply test/manager.js]: Done.

**Comment by wswoodruff** [comment]
Thanks @sholladay!

**Pull request merged**